### PR TITLE
further refactor shock_moments subroutine

### PR DESCRIPTION
**Pull request by xyuan**
further refactor shock_moments subroutine

[BFB]

### CODE REVIEW

**Comment by bogensch** [comment]
Thanks for doing this.  I realize though that the comments I left on issue #387 were fairly ambiguous and these changes do not fully reflect what my intentions were (my fault).  I will make some further changes and commit them to this branch for review.

**Comment by xyuan** [comment]
Alright, thank you very much.

**Comment by bogensch** [comment]
My modifications have been added and ready for review.  Mostly I added two subroutines, "calc_shoc_varorcovar" in which thl_sec, qw_sec, and qwthl_sec are now all calculated using the same routine and "calc_shoc_vertflux" in which wthl_sec, qw_sec, uw_sec, vw_sec, wtke_sec, and wtracer_sec all now share the same routine.  

Note that this PR produces round off differences, thus the SHOC run and compare test is expected to fail.  This is because previously the thl_sec used to be computed by (thetal(i,kt)-thetal(i,k))**2 and is now computed by (thetal(i,kt)-thetal(i,k))*(thetal(i,kt)-thetal(i,k)), which produces round off differences since the calculation is not done the exact same way.  Note that I verified this by swapping out a new routine for thl_sec which retains the old formulation that did produce b4b.

**Comment by xyuan** [comment]
@pressel could you take some time to review this PR?

File: components/cam/src/physics/cam/shoc.F90 [hunk 1]
```diff
@@ -1127,9 +1127,9 @@ subroutine diag_second_shoc_moments(&
   real(rtype), intent(in) :: wthl_sfc(shcol)
   ! Surface latent heat flux [kg/kg m/s]
   real(rtype), intent(in) :: wqw_sfc(shcol)
-  ! Surface momentum flux (u-direction) [m3/s3]
+  ! Surface momentum flux (u-direction) [m2/s2]
```
Comment (pressel) [inline components/cam/src/physics/cam/shoc.F90:1130]: Nice catch on the error in units.

**Comment by xyuan** [comment]
@bogensch could we manually merge this PR since this is non-BFB?

**Comment by PeterCaldwell** [comment]
@xyuan and @bogensch - is this PR ready to be hand-merged? If you say so, Jim or I will merge it.

**Comment by bogensch** [comment]
yes, @PeterCaldwell it is ready.  thanks!

**Comment by PeterCaldwell** [comment]
Ok, done.

**Pull request merged**